recipes/libcurl/all/conanfile.py

```diff
@@ -207,13 +204,22 @@ def build_requirements(self):
         else:
             self.build_requires("libtool/2.4.6")
             self.build_requires("pkgconf/1.7.4")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            if self._settings_build.os == "Windows" and not get_env("CONAN_BASH_PATH"):
                 self.build_requires("msys2/cci.latest")
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-        tools.download("https://curl.haxx.se/ca/cacert.pem", "cacert.pem", verify=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+        download(self, "https://curl.haxx.se/ca/cacert.pem", "cacert.pem", verify=True)
+
+    def generate(self):
+        if self._is_using_cmake_build:
+            self._generate_with_cmake()
+        else:
+            self._generate_with_autotools()
```

Comment:
```suggestion
            self._generate_with_autotools()
        ms = VirtualBuildEnv(self)
        ms.generate(scope="build")
```
Needed because there is tool_requires. Also, don't forget to import VirtualBuildEnv